### PR TITLE
fix(module-resolver): resolve with `config.rootDir` when defined

### PR DESCRIPTION
**Pull request by blittle**
## Details
With the new module resolution implementation, `config.rootDir` was removed for resolving the configuration. This is the old implementation: https://github.com/salesforce/lwc/blob/ce73c8e906fc9aba884b520d87b6191332dc4f11/packages/%40lwc/module-resolver/src/index.ts#L117

In particular this broke projects with `npm link` dependencies.

## Does this PR introduce breaking changes?

* ✅ `No, it does not introduce breaking changes.`

If yes, please describe the impact and migration path for existing applications.

## The PR fulfills these requirements:
* Have tests for the proposed changes been added? ✅
* Have you followed [these instructions](../CONTRIBUTING.md#-commit-message-conventions) to clearly describe the issue being fixed or feature enhanced? ✅

### CODE REVIEW

File: packages/@lwc/module-resolver/src/__tests__/fixtures/linked/module-a/node_modules/deps [hunk 1]
```diff
@@ -0,0 +1 @@
+../../module-b
```
Comment (blittle) [inline packages/@lwc/module-resolver/src/__tests__/fixtures/linked/module-a/node_modules/deps]: This is a symlink

File: packages/@lwc/module-resolver/src/index.ts [hunk 2]
```diff
@@ -202,8 +202,8 @@ export function resolveModule(
             `The importee argument must be a valid LWC module name. Received "${importee}"`
         );
     }
-
-    const rootDir = findFirstUpwardConfigPath(path.resolve(dirname));
+    const customRootDir = config && config.rootDir ? config.rootDir : null;
+    const rootDir = findFirstUpwardConfigPath(path.resolve(customRootDir || dirname));
```
Comment (blittle) [inline packages/@lwc/module-resolver/src/index.ts]: I think this matches the old functionality of `rootDir`: https://github.com/salesforce/lwc/blob/ce73c8e906fc9aba884b520d87b6191332dc4f11/packages/%40lwc/module-resolver/src/index.ts#L117

**Comment by blittle** [comment]
Looks like a rollup test is broken. Looking into it.

**Comment by blittle** [comment]
This repo reproduces the problem: https://github.com/blittle/rootdir-example

**Comment by diervo** [comment]
Closing this since this is an issue with the linking process.

**Pull request closed without merging**